### PR TITLE
feat: add union and interface support

### PR DESCRIPTION
**Pull request by PacoDu**
This PR adds support for union and interface, and related tests !

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -108,6 +110,10 @@ module.exports = fp(async function (app, opts) {
       } else if (type instanceof GraphQLObjectType) {
         const fields = type.getFields()
         const resolver = resolvers[name]
+        if (resolver.__isTypeOf) {
+          type.isTypeOf = resolver.__isTypeOf
+          delete resolver.__isTypeOf
+        }
```
Comment (mcollina) [inline index.js]: What are these lines for? What is `__isTypeOf`?

File: test/app-decorator.js [hunk 2]
```diff
@@ -464,3 +464,282 @@ test('enum should be supported', async (t) => {
...
+      },
+      getGeometryMultiPolygon: async () => {
+        return {
+          type: 'MultiPolygon',
+          coordinates: 1
+        }
+      }
+    },
+    Geometry: {
+      __resolveType (geometry) {
```
Comment (mcollina) [inline test/app-decorator.js]: What is `__resolveType`?

File: index.js [hunk 3]
```diff
@@ -116,6 +122,9 @@ module.exports = fp(async function (app, opts) {
         for (const prop of Object.keys(resolver)) {
           type[prop] = resolver[prop]
         }
+      } else if (type instanceof GraphQLInterfaceType || type instanceof GraphQLUnionType) {
+        const resolver = resolvers[name]
+        type.resolveType = resolver.__resolveType
```
Comment (mcollina) [inline index.js]: Is this common syntax in the graphql world?

Comment (PacoDu) [reply index.js]: You are right this is not part of the official `graphql` lib, I confused with Apollo's makeExecutableSchema which use `__` syntax. 

I was looking for a way to add union and interface without using `graphql-tool` or raw schema definition. But I've mixed both libraries ... 👎 

I will remove the `__` to match the official `graphql` library, does it suits you ? 
Adding `isTypeOf` and `resolveType` will allow users to define resolvers for union and interface defined in the schema definition.

Comment (mcollina) [reply index.js]: Awesome!

**Pull request merged**